cypress/support/step_definitions/dashboard/there_is_a_list_with_all_states.js

```diff
@@ -29,53 +31,71 @@ function createListForService(service) {
       where: {
         listItem: {
           list: {
-            reference: "SMOKE",
+            reference,
           },
         },
       },
     },
   });
 
   cy.task("db", {
-    operation: "list.upsert",
+    operation: "user.findUnique",
     variables: {
-      create: {
-        type: service,
-        reference: "SMOKE",
-        nextAnnualReviewStartDate: null,
-        jsonData,
-        country: {
-          connect: {
-            name: "Eurasia",
+      where: {
+        email: "smoke@cautionyourblast.com",
+      },
+    },
+  }).then(result => {
+    cy.task("db", {
+      operation: "list.upsert",
+      variables: {
+        create: {
+          type: service,
+          reference,
+          nextAnnualReviewStartDate: null,
+          jsonData,
+          country: {
+            connect: {
+              name: "Eurasia",
+            },
           },
+          users: {
+            connect: { id: result.id }
+          }
         },
-      },
-      update: {
-        type: service,
-        jsonData,
-        nextAnnualReviewStartDate: null,
-        items: {
-          deleteMany: {},
+        update: {
+          type: service,
+          jsonData,
+          nextAnnualReviewStartDate: null,
+          items: {
+            deleteMany: {},
+          },
+          users: {
+            connect: { id: result.id }
+          }
+        },
+        where: {
+          reference,
         },
       },
-      where: {
-        reference: "SMOKE",
-      },
-    },
+    });
   });
+
 }
 
 Given("there are these list items", (table) => {
   /**
    * | contactName | companyName | status | isPublished |  emailVerified | isPinned | displayedRadioButtons | hiddenRadioButtons | service | isArchived
    */
   const rows = table.hashes();
+  const { service } = rows.at(0);
+  const reference = service === "funeralDirectors" ? "SMOKE-FD" : "SMOKE";
```

Comment:
I had an error when the `SMOKE-FD` list already existed in my database which is a funeral director for Eurasia. 

Then the test runs it tries to add another FD list for Eurasia under the reference `SMOKE` but fails because there's already an FD list for said country.

So I changed the code so that it only adds an FD list with the reference `SMOKE-FD`.

Reply:
So this file `cypress/support/step_definitions/dashboard/users/related-links/a_related_list_exists.js` has a hardcoded 'SMOKE-FD' reference and I think this had run at some point so the list was already in the DB when I ran the other tests.

I guess I could have changed this to 'SMOKE' form 'SMOKE-FD' to fix this but 🤷